### PR TITLE
block domain tiny.realdevsquad.com while creating tiny url

### PR DESCRIPTION
**Pull request by fakhruddinkw**
## Issue Ticket Number

This PR takes care of #131 

## Description

Changes in this PR will not let a user create a tiny url of the active tiny url

**Breaking Changes**

-   [ ] Yes
-   [x] No

_If your feature introduces breaking changes or if something is missing, please mention the related issue tickets._

**Development Tested?**

-   [x] Yes
-   [ ] No

_Confirm whether the changes have been tested locally during development._

**Tested in Staging?**

-   [ ] Yes
-   [x] No

_Indicate whether the changes have been tested in the staging environment for dev to main._

**Database Changes**

-   [ ] Yes
-   [x] No

_Indicate whether the changes include modifications to the database._

### Screenshots

https://github.com/user-attachments/assets/8b716b87-6e33-4980-8e8b-9dcf991af131

## Additional Notes

Include any additional notes, considerations, or explanations that might be helpful for reviewers.

### CODE REVIEW

**Comment by vinit717** [comment]
Can you pass me the design doc so I can get the what approach you are following

**Comment by iamitprakash** [comment]
*please remove logic from controller *

**Comment by fakhruddinkw** [comment]
> Can you pass me the design doc so I can get the what approach you are following

As I believed this was a straight solution, I didn't make any design doc. However I did go through the doc which sebin (last assignee) made.

File: controllers/url.go [hunk 1]
```diff
@@ -92,12 +93,33 @@ func CreateTinyURL(ctx *gin.Context, db *bun.DB) {
...
+	}
+
+	if strings.Contains(strings.ToLower(parsedUrl.Host),strings.ToLower(config.Domain)) {
+			
+		path :=parsedUrl.Path
+	
+		if len(path) >=1  {
+
+			ctx.JSON(http.StatusForbidden, dtos.URLCreationResponse{
+				Message: "Cannot create tiny url of tiny url!",
```
Comment (yesyash) [inline controllers/url.go:117]:
```suggestion
				Message: "Cannot create a tiny url for tiny.realdevsquad.com",
```

**Review by yesyash — CHANGES_REQUESTED** [review]

File: controllers/url.go (hunk 1 above)
Comment (fakhruddinkw) [reply controllers/url.go]: Done!

File: controllers/url.go [hunk 2]
```diff
@@ -92,12 +93,33 @@ func CreateTinyURL(ctx *gin.Context, db *bun.DB) {
...
+	}
+
+	if strings.Contains(strings.ToLower(parsedUrl.Host),strings.ToLower(config.Domain)) {
+			
+		path :=parsedUrl.Path
+	
+		if len(path) >=1  {
+
+			ctx.JSON(http.StatusForbidden, dtos.URLCreationResponse{
+				Message: "Cannot create a tiny url for tiny.realdevsquad.com",
```
Comment (yesyash) [inline controllers/url.go:117]: - can we use config.Domain instead of hardcoding `tiny.realdevsquad.com` to get the correct error message on staging?

**Review by yesyash — CHANGES_REQUESTED** [review]

File: controllers/url.go (hunk 2 above)
Comment (fakhruddinkw) [reply controllers/url.go:117]: great suggestion. Working on it. Thank you!

**Review by vinit717 — APPROVED** [review]
Maintain the follow up for enhancement

**Comment by fakhruddinkw** [comment]
> Maintain the follow up for enhancement

picking it up right after this! thank you

**Review by yesyash — APPROVED** [review]

**Pull request merged**